### PR TITLE
#65 Changed backend

### PR DESCRIPTION
**Pull request by MaximSmolskiy**

### CODE REVIEW

File: app/src/main/java/com/mithridat/nonoconverter/backend/Field.java [hunk 1]
```diff
@@ -24,7 +29,7 @@
      * @param rows - number of field rows
      * @param cols - number of field columns
      */
-    Field(int rows, int cols) {
+    public Field(int rows, int cols) {
         _field = new int[rows][cols];
         for(int i = 0; i < rows; i++) {
             for(int j = 0; j < cols; j++) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/backend/Field.java:35]: Please initialize the matrix with `EMPTY` value

File: app/src/main/java/com/mithridat/nonoconverter/backend/solver/NonogramSolver.java [hunk 2]
```diff
@@ -89,10 +86,25 @@ private void init() {
...
      *         false, otherwise
      */
-    private boolean check() {
+    boolean check() {
         return true;
     }
 
+    /**
+     * Method for getting workspace in row or column:
+     * - each row contains start position and next to last position
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/backend/solver/NonogramSolver.java]: Maybe `each row of result matrix`?

File: app/src/main/java/com/mithridat/nonoconverter/backend/solver/Nonogram.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+     * with column number i and row number j
+     *
+     * @param i - column number
+     * @param j - row number
+     * @return top nonogram grid cell with column number i and row number j
+     */
+    int getTopColValue(int i, int j) {
+        return _top[i][j];
+    }
+
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/backend/solver/Nonogram.java:100]: Perhaps there are not methods for getting `_left[i]` and `_top[i]` lengths.

**Review by maik-nack — APPROVED** [review]
Looks Great!

**Review by PawelMelnik — APPROVED** [review]
Looks ok.

**Pull request merged**